opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1053,7 +1053,7 @@ namespace Opm
         auto& ws = well_state.well(this->index_of_well_);
         int nonzero_rate_index = -1;
         for (int p = 0; p < this->number_of_phases_; ++p) {
-            if (ws.surface_rates[p] != 0.0) {
+            if (std::abs(ws.surface_rates[p]) > 1e-14) {
```

Comment:
That is a very low epsilon, I am surprised it had an observable effect! It also seems somewhat randomly chosen? As this is intended to account for calculation differences that do not depend on the magnitude of rates but on floating point errors etc. (at least that is how I interpret it), perhaps we should give it a name indicating this, such as `floating_point_error_epsilon`?

Reply:
It is sufficiently large to fix the issue I encountered, but we may want to consider make it a bit larger if we get into convergence issues for wells with very small rates. Lets cross that bridge when we get there...